modules/data.atmosphere/R/ECMWF_helper_functions.R

```diff
@@ -0,0 +1,283 @@
+#' Helper functions for met2CF.ECMWF
+#' 
+#' mergenc_ECMWF writes 1 CF and 50 PF netCDF files by extracting data from GRIB2 files and filling NaNs at 3h position wherever data is unavailable. 
+#'
+#' @param lat latitude
+#' @param lon longitude
+#' @param outfolder Path to directory where nc files need to be saved.
+#' @param overwrite Logical if files needs to be overwritten.
+#' @param verbose Logical flag defining if output of function be extra verbose.
+#'
+#' @return list of dataframes
+#' @export
+#' 
+#' \dontrun{
+#' results <- mergenc_ECMWF(lat.in,
+#'                          lon.in,
+#'                          outfolder,
+#'                          overwrite = overwrite)
+#'
+#' @author Swarnalee Mazumder
+#' 
+nanfill <- function(variable) { # nanfill function fills NaN at every 3h period from 144h to 360h (original step 6h)
+  
+  len6 <- variable
+  len3nan <- rep(NA,length(len6)*2-1)
+  len3nan[seq(1, length(len6)*2-1, 2)]  <- len6
+  
+  return(len3nan)
+}
+
+mergenc_ECMWF <- function(lat.in,
+                          lon.in,
+                          outfolder,
+                          overwrite = FALSE,
+                          verbose = TRUE) {
+  
+  # Python script "download_ecmwf.py" to get latest forecast date and download forecast datasets
+  script.path = file.path(system.file("ECMWF/download_ecmwf.py", package = "PEcAn.data.atmosphere"))
+  reticulate::source_python(script.path)
+  
+  time <- 0
+  stream <- "enfo"
+  type <- c("cf", "pf")
+  step_15day <- 360
+  
+  all_parameters <-  c("10u", "10v", "2t", "sp", "tp")
+  
+  date_latestdata <- ecmwflatest(time= time, step15= step_15day, stream= stream, type= type, params= all_parameters)
+  
+  latest_filedate <- strtoi(gsub("-", "", substr(as.character.Date(date_latestdata), 1, 10)))
+  
+  current_date <- strtoi(gsub("-", "", Sys.Date()))
+  
+  in_fname <- paste(latest_filedate, time, stream, sep = "_")
+  
+  script.path = file.path(system.file("ECMWF/met2CFutils.py", package = "PEcAn.data.atmosphere"))
+  reticulate::source_python(script.path)
+  
+  fnames_3h <- all_filenames_3h(in_fname)
+  fnames_6h <- all_filenames_6h(in_fname)
+  
+  ######### CF ######### 
+  
+  ##### 3h CF -> 0 - 141h at 3h timestep
+  cf_3h_out_fname <- paste0(latest_filedate, time, stream, "cf_3h", sep = "_")
+  cf_3h_nc <- combine_files_cf(fnames_3h, lat_in, lon_in, out_filename= cf_3h_out_fname)
```

Comment:
I thought we agreed to not write these intermediate 3hrly and 6 hrly netcdf files, are these still being written?